src/FSharpPlus/Validations.fs

```diff
@@ -0,0 +1,137 @@
+module FSharpPlus.Validations
+
+open FSharpPlus
+open FSharpPlus.Lens
+open FSharpPlus.Data
+
+/// An 'AccValidation' is either a value of the type 'err or 'a, similar to 'Result'. However,
+/// the 'Applicative' instance for 'AccValidation' /accumulates/ errors using a 'Semigroup' on 'err.
+/// In contrast, the Applicative for 'Result' returns only the first error.
+///
+/// A consequence of this is that 'AccValidation' is not a monad. There is no F#+ 'Bind' method since
+/// that would violate monad rules.
+///
+/// An example of typical usage can be found <https://github.com/qfpl/validation/blob/master/examples/src/Email.hs here>.
+///
+type AccValidation<'err,'a> =
+  | AccFailure of 'err
+  | AccSuccess of 'a
+
+module AccValidation=
+  let inline map (f:'T->'U)=function
+    |AccFailure e -> AccFailure e
+    |AccSuccess a -> AccSuccess (f a) 
+  let inline apply e1' e2' = 
+    match e1',e2' with
+    |AccFailure e1, AccFailure e2 -> AccFailure (plus e1 e2)
+    |AccFailure e1, AccSuccess _  -> AccFailure e1
+    |AccSuccess _, AccFailure e2 -> AccFailure e2
+    |AccSuccess f, AccSuccess a -> AccSuccess (f a)
+  let inline foldr f x = function 
```

Comment:
This should be ``foldBack`` I think I missed it in my dehaskellize PR. Note that the signature (params order) of Haskell's ``foldr`` is not the same as ``foldBack``.